src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc

```diff
@@ -86,9 +93,29 @@
   [router]
   (set (comp/component-options router :router-targets)))
 
-(defn route-immediate [ident] (with-meta ident {:immediate true}))
-(defn route-deferred [ident completion-fn] (with-meta ident {:immediate false
-                                                             :fn        completion-fn}))
+(defn- ident-matches-expectation? [[expected-table maybe-expected-id] [table id]]
+  ;; NOTE: If the `id` of the ident is hardcoded then maybe-expected-id will be set,
+  ;; but if it depends on props then it will be nil
+  (and (= expected-table table)
+       (or (nil? maybe-expected-id) (= maybe-expected-id id))))
+
+(defn- check-ident-matches-expectation? [fn-name ident]
+  (when (and #?(:clj false :cljs goog.DEBUG)
+             *target-class*
+             (not (ident-matches-expectation? (comp/ident *target-class* {}) ident)))
+    (log/warn fn-name " was invoked with the ident " ident
```

Comment:
should this be a warning or rather an error? Perhaps error?

Reply:
yeah, error. You're already allowing for the possible nil id, so the only reason you would be logging is if it really looks like an error.